### DISCUSSION TITLE
How to Build for Pico2?

### DISCUSSION THREAD

**Discussion by thestumbler** [micropython/micropython#15690] [RP2040 / Pico]
I see in the PR for the Pico 2 RP2350 that the board name is undecided. Does that mean we can't build it locally yet? I think I've pulled the latest and greatest MicroPython but I can't find any board whose name suggests it's the RP2350.

**Comment by mattytrentini** [comment]
In the PR (#[15619](https://github.com/micropython/micropython/pull/15619)) the board name is currently "RPI_PICO2" and builds cleanly using that name.

The comment was just raising the issue that the board name may yet change.

**Reply by thestumbler** [reply]
Thanks, it's working. I guess what confused me is there is no corresponding RPI_PICO2 configuration folder in the BOARDS directory. And when I build using `board=RPI_PICO2` it goes in the same build directory as `board=RPI_PICO`.

**Reply by thestumbler** [reply]
slight correction ... I can now build it, but the resulting binary isn't working. But I attribute this to the fact that I'm using a Pimoroni board and don't have a real Pico 2 yet.

**Reply by thestumbler** [reply]
And, I figured it out. I don't know how to pull a not-yet-merged pull request. I will just wait until it is officially merged.

**Reply by robert-hh** [reply]
> I don't know how to pull a not-yet-merged pull request. 

Assuming that the MP repository is called `origin` and the new branches name shall be `rp2-add-2350`:

```
git fetch origin pull/15619/head:rp2-add-2350
git checkout rp2-add-2350
git submodule sync
git submodule update --init
```

**Reply by thestumbler** [reply]
Well, I almost succeeded by a similar tact — just pulling the whole PR branch from D P George’s repository. That almost worked, but it seemed to be failing because of some submodules and toolkits needing to be updated. I briefly tried the obvious things, but eventually gave up. This experiment was just a curiosity, as I have no real need to build it from scratch. That’s just the procedure I’ve always used for MP, either my own custom boards or standard off-the-shelf ones. I’ll just wait for this PR to be merged into the main MP repo. 

BTW, Pimiromi has made some board definition files for their new RP2350 boards over in their repository. I fumbled a bit trying to make them work, and no joy. But it’s clear that when all the dust settles, it seems trivial to roll in their new modules into the boards directory

**Reply by mattytrentini** [reply]
An alternative is to use [GitHub CLI](https://cli.github.com/) which helps with a number of operations, including pulling and switching branches by PR number:

```bash
> gh pr checkout 15619
```

**Reply by thestumbler** [reply]
Thanks Matt. I discovered that command and initially it didn’t seem to work for me (in that it built but didn’t make a binary that worked). In hindsight I think it would have been just fine if I had followed through with the submodule updates.

**Comment by robert-hh** [comment]
You have to add:

git submodule sync; git submodule update --init

**Reply by thestumbler** [reply]
I thought I did that, but I’ll try again later tonight. Thanks for the tips.
